Platform/ResetNvramEntry/ResetNvramEntry.c

```diff
@@ -0,0 +1,168 @@
+/** @file
+  Boot entry protocol implementation of Reset NVRAM boot picker entry.
+
+  Copyright (c) 2022, Mike Beaton. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-3-Clause
+**/
+
+#include <Guid/AppleVariable.h>
+
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/OcDeviceMiscLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+
+#include <Protocol/OcBootEntry.h>
+
+#define OC_MENU_RESET_NVRAM_ENTRY  "Reset NVRAM"
+
+STATIC BOOLEAN  mIsNative     = FALSE;
+STATIC BOOLEAN  mPreserveBoot = FALSE;
+
+STATIC OC_PICKER_ENTRY  mResetNvramBootEntries[1] = {
+  {
+    .Id            = "reset_nvram",
+    .Name          = OC_MENU_RESET_NVRAM_ENTRY,
+    .Path          = NULL,
+    .Arguments     = NULL,
+    .Flavour       = OC_FLAVOUR_RESET_NVRAM,
+    .Auxiliary     = TRUE,
+    .Tool          = FALSE,
+    .TextMode      = FALSE,
+    .RealPath      = FALSE,
+    .SystemAction  = NULL, ///< overridden in set up
+    .ActionConfig  = &mPreserveBoot,
+    .AudioBasePath = OC_VOICE_OVER_AUDIO_FILE_RESET_NVRAM,
+    .AudioBaseType = OC_VOICE_OVER_AUDIO_BASE_TYPE_OPEN_CORE
+  }
+};
+
+STATIC
+EFI_STATUS
+EFIAPI
+ResetNvramGetBootEntries (
+  IN           OC_PICKER_CONTEXT  *PickerContext,
+  IN     CONST EFI_HANDLE         Device,
+  OUT       OC_PICKER_ENTRY       **Entries,
+  OUT       UINTN                 *NumEntries
+  )
+{
+  //
+  // Custom entries only.
+  //
+  if (Device != NULL) {
```

Comment:
Hmm, UEFI has the convention to NULL-check all used parameters in protocol calls. However, as it's our own protocol, we could allow this. I never really understood that convention anyway. Cc @vit9696

Reply:
Well, that was not the point, it was about the arguments that do _not_ get NULL-checked.

Reply:
I think we should follow suit and return EFI_INVALID_PARAMETER for all non-optional NULL pointer arguments.

Reply:
So, if I mark this OPTIONAL, that's all I need to do here, right?